### PR TITLE
Update cgmanifest to include vcpkg dependencies and linux tools we use in CI.

### PR DESCRIPTION
**Pull request by ahsonkhan**
Documenting all `AptDependencies` and `vcpkg.deps`

**These, currently, include:**
- curl (which transitively depends on openssl, and zlib)
- paho-mqtt
- cmocka
- gcc-arm-none-eabi
- gcovr
- lcov
- clang-format
- doxygen

### CODE REVIEW

File: eng/pipelines/templates/jobs/archetype-sdk-client.yml [hunk 1]
```diff
@@ -343,6 +343,22 @@ jobs:
       # Run clang-format recursively on each source and header file within the repo.
       find . \( -iname '*.h' -o -iname '*.c' \) -exec clang-format -i {} \;
       
+      echo CLANG FORMAT INFO
```
Comment (ahsonkhan) [inline eng/pipelines/templates/jobs/archetype-sdk-client.yml:346]: Unrelated change which I will revert. Just using it to verify version numbers.

**Review by antkmsft — APPROVED** [review]
I see we are freezing the version numbers which is the best approach for build stability. However, what is our plan to upgrade these dependencies, do we have a process, cadence, or is there an event that would trigger us to go and grab more latest versions?

File: cgmanifest.json [hunk 2]
```diff
@@ -19,6 +19,116 @@
...
+    {
+      "Component": {
+        "Type": "other",
+        "Other": {
+          "Name": "cmocka",
+          "Version": "1.1.5-1",
+          "DownloadUrl": "https://gitlab.com/cmocka/cmocka"
+        }
+      },
+      "DevelopmentDependency": false
```
Comment (RickWinter) [inline cgmanifest.json:32]: Are we including parts of CMocka outside of the tests?  Otherwise I'd expect this one is true that its a dev only dependency

**Review by RickWinter — APPROVED** [review]

**Comment by ahsonkhan** [comment]
> I see we are freezing the version numbers which is the best approach for build stability.

This file and PR in itself doesn't freeze anything (that isn't the purpose of cgmanifest, all we are doing is declaring our dependencies). We rely on vcpkg to handle the versioning for us. The other versions are defaults that we get on our build images based on OSes like ubuntu-18.04, etc. when installed.

We pinned the vcpkg version to this:
https://github.com/Azure/azure-sdk-for-c/blob/fad6ccb651a438ec6214d3d4944c051f72f71630/eng/vcpkg-commit.txt#L1

**Review by vhvb1989 — APPROVED** [review]
Great work Sr.! Thank you

**Pull request merged**